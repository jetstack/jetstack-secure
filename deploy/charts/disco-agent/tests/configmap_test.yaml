suite: test the contents of the config.yaml
templates:
  - configmap.yaml
release:
  name: test
  namespace: test-ns
tests:
  - it: defaults
    asserts:
      - matchSnapshot: {}

  - it: custom-period
    set:
      config.period: 1m
    asserts:
      - matchSnapshot: {}
