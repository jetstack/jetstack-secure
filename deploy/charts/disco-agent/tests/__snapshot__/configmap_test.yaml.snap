custom-cluster-description:
  1: |
    apiVersion: v1
    data:
      config.yaml: |-
        cluster_name: ""
        cluster_description: "A cloud hosted Kubernetes cluster hosting production workloads.\n\nteam: team-1\nemail: team-1@example.com\npurpose: Production workloads\n"
        period: "12h0m0s"
        data-gatherers:
        - kind: k8s-discovery
          name: ark/discovery
        - kind: k8s-dynamic
          name: ark/secrets
          config:
            resource-type:
              version: v1
              resource: secrets
            field-selectors:
            - type!=kubernetes.io/dockercfg
            - type!=kubernetes.io/dockerconfigjson
            - type!=bootstrap.kubernetes.io/token
            - type!=helm.sh/release.v1
        - kind: k8s-dynamic
          name: ark/serviceaccounts
          config:
            resource-type:
              resource: serviceaccounts
              version: v1
        - kind: k8s-dynamic
          name: ark/roles
          config:
            resource-type:
              version: v1
              group: rbac.authorization.k8s.io
              resource: roles
        - kind: k8s-dynamic
          name: ark/clusterroles
          config:
            resource-type:
              version: v1
              group: rbac.authorization.k8s.io
              resource: clusterroles
        - kind: k8s-dynamic
          name: ark/rolebindings
          config:
            resource-type:
              version: v1
              group: rbac.authorization.k8s.io
              resource: rolebindings
        - kind: k8s-dynamic
          name: ark/clusterrolebindings
          config:
            resource-type:
              version: v1
              group: rbac.authorization.k8s.io
              resource: clusterrolebindings
        - kind: k8s-dynamic
          name: ark/jobs
          config:
            resource-type:
              version: v1
              group: batch
              resource: jobs
        - kind: k8s-dynamic
          name: ark/cronjobs
          config:
            resource-type:
              version: v1
              group: batch
              resource: cronjobs
        - kind: k8s-dynamic
          name: ark/deployments
          config:
            resource-type:
              version: v1
              group: apps
              resource: deployments
        - kind: k8s-dynamic
          name: ark/statefulsets
          config:
            resource-type:
              version: v1
              group: apps
              resource: statefulsets
        - kind: k8s-dynamic
          name: ark/daemonsets
          config:
            resource-type:
              version: v1
              group: apps
              resource: daemonsets
        - kind: k8s-dynamic
          name: ark/pods
          config:
            resource-type:
              version: v1
              resource: pods
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: disco-agent
        app.kubernetes.io/version: v0.0.0
        helm.sh/chart: disco-agent-0.0.0
      name: test-disco-agent-config
      namespace: test-ns
custom-cluster-name:
  1: |
    apiVersion: v1
    data:
      config.yaml: |-
        cluster_name: "cluster-1 region-1 cloud-1 "
        cluster_description: ""
        period: "12h0m0s"
        data-gatherers:
        - kind: k8s-discovery
          name: ark/discovery
        - kind: k8s-dynamic
          name: ark/secrets
          config:
            resource-type:
              version: v1
              resource: secrets
            field-selectors:
            - type!=kubernetes.io/dockercfg
            - type!=kubernetes.io/dockerconfigjson
            - type!=bootstrap.kubernetes.io/token
            - type!=helm.sh/release.v1
        - kind: k8s-dynamic
          name: ark/serviceaccounts
          config:
            resource-type:
              resource: serviceaccounts
              version: v1
        - kind: k8s-dynamic
          name: ark/roles
          config:
            resource-type:
              version: v1
              group: rbac.authorization.k8s.io
              resource: roles
        - kind: k8s-dynamic
          name: ark/clusterroles
          config:
            resource-type:
              version: v1
              group: rbac.authorization.k8s.io
              resource: clusterroles
        - kind: k8s-dynamic
          name: ark/rolebindings
          config:
            resource-type:
              version: v1
              group: rbac.authorization.k8s.io
              resource: rolebindings
        - kind: k8s-dynamic
          name: ark/clusterrolebindings
          config:
            resource-type:
              version: v1
              group: rbac.authorization.k8s.io
              resource: clusterrolebindings
        - kind: k8s-dynamic
          name: ark/jobs
          config:
            resource-type:
              version: v1
              group: batch
              resource: jobs
        - kind: k8s-dynamic
          name: ark/cronjobs
          config:
            resource-type:
              version: v1
              group: batch
              resource: cronjobs
        - kind: k8s-dynamic
          name: ark/deployments
          config:
            resource-type:
              version: v1
              group: apps
              resource: deployments
        - kind: k8s-dynamic
          name: ark/statefulsets
          config:
            resource-type:
              version: v1
              group: apps
              resource: statefulsets
        - kind: k8s-dynamic
          name: ark/daemonsets
          config:
            resource-type:
              version: v1
              group: apps
              resource: daemonsets
        - kind: k8s-dynamic
          name: ark/pods
          config:
            resource-type:
              version: v1
              resource: pods
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: disco-agent
        app.kubernetes.io/version: v0.0.0
        helm.sh/chart: disco-agent-0.0.0
      name: test-disco-agent-config
      namespace: test-ns
custom-period:
  1: |
    apiVersion: v1
    data:
      config.yaml: |-
        cluster_name: ""
        cluster_description: ""
        period: "1m"
        data-gatherers:
        - kind: k8s-discovery
          name: ark/discovery
        - kind: k8s-dynamic
          name: ark/secrets
          config:
            resource-type:
              version: v1
              resource: secrets
            field-selectors:
            - type!=kubernetes.io/dockercfg
            - type!=kubernetes.io/dockerconfigjson
            - type!=bootstrap.kubernetes.io/token
            - type!=helm.sh/release.v1
        - kind: k8s-dynamic
          name: ark/serviceaccounts
          config:
            resource-type:
              resource: serviceaccounts
              version: v1
        - kind: k8s-dynamic
          name: ark/roles
          config:
            resource-type:
              version: v1
              group: rbac.authorization.k8s.io
              resource: roles
        - kind: k8s-dynamic
          name: ark/clusterroles
          config:
            resource-type:
              version: v1
              group: rbac.authorization.k8s.io
              resource: clusterroles
        - kind: k8s-dynamic
          name: ark/rolebindings
          config:
            resource-type:
              version: v1
              group: rbac.authorization.k8s.io
              resource: rolebindings
        - kind: k8s-dynamic
          name: ark/clusterrolebindings
          config:
            resource-type:
              version: v1
              group: rbac.authorization.k8s.io
              resource: clusterrolebindings
        - kind: k8s-dynamic
          name: ark/jobs
          config:
            resource-type:
              version: v1
              group: batch
              resource: jobs
        - kind: k8s-dynamic
          name: ark/cronjobs
          config:
            resource-type:
              version: v1
              group: batch
              resource: cronjobs
        - kind: k8s-dynamic
          name: ark/deployments
          config:
            resource-type:
              version: v1
              group: apps
              resource: deployments
        - kind: k8s-dynamic
          name: ark/statefulsets
          config:
            resource-type:
              version: v1
              group: apps
              resource: statefulsets
        - kind: k8s-dynamic
          name: ark/daemonsets
          config:
            resource-type:
              version: v1
              group: apps
              resource: daemonsets
        - kind: k8s-dynamic
          name: ark/pods
          config:
            resource-type:
              version: v1
              resource: pods
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: disco-agent
        app.kubernetes.io/version: v0.0.0
        helm.sh/chart: disco-agent-0.0.0
      name: test-disco-agent-config
      namespace: test-ns
defaults:
  1: |
    apiVersion: v1
    data:
      config.yaml: |-
        cluster_name: ""
        cluster_description: ""
        period: "12h0m0s"
        data-gatherers:
        - kind: k8s-discovery
          name: ark/discovery
        - kind: k8s-dynamic
          name: ark/secrets
          config:
            resource-type:
              version: v1
              resource: secrets
            field-selectors:
            - type!=kubernetes.io/dockercfg
            - type!=kubernetes.io/dockerconfigjson
            - type!=bootstrap.kubernetes.io/token
            - type!=helm.sh/release.v1
        - kind: k8s-dynamic
          name: ark/serviceaccounts
          config:
            resource-type:
              resource: serviceaccounts
              version: v1
        - kind: k8s-dynamic
          name: ark/roles
          config:
            resource-type:
              version: v1
              group: rbac.authorization.k8s.io
              resource: roles
        - kind: k8s-dynamic
          name: ark/clusterroles
          config:
            resource-type:
              version: v1
              group: rbac.authorization.k8s.io
              resource: clusterroles
        - kind: k8s-dynamic
          name: ark/rolebindings
          config:
            resource-type:
              version: v1
              group: rbac.authorization.k8s.io
              resource: rolebindings
        - kind: k8s-dynamic
          name: ark/clusterrolebindings
          config:
            resource-type:
              version: v1
              group: rbac.authorization.k8s.io
              resource: clusterrolebindings
        - kind: k8s-dynamic
          name: ark/jobs
          config:
            resource-type:
              version: v1
              group: batch
              resource: jobs
        - kind: k8s-dynamic
          name: ark/cronjobs
          config:
            resource-type:
              version: v1
              group: batch
              resource: cronjobs
        - kind: k8s-dynamic
          name: ark/deployments
          config:
            resource-type:
              version: v1
              group: apps
              resource: deployments
        - kind: k8s-dynamic
          name: ark/statefulsets
          config:
            resource-type:
              version: v1
              group: apps
              resource: statefulsets
        - kind: k8s-dynamic
          name: ark/daemonsets
          config:
            resource-type:
              version: v1
              group: apps
              resource: daemonsets
        - kind: k8s-dynamic
          name: ark/pods
          config:
            resource-type:
              version: v1
              resource: pods
    kind: ConfigMap
    metadata:
      labels:
        app.kubernetes.io/instance: test
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: disco-agent
        app.kubernetes.io/version: v0.0.0
        helm.sh/chart: disco-agent-0.0.0
      name: test-disco-agent-config
      namespace: test-ns
